File: Ejercicios_POO3.1_Java/src/Ejercicio3_JuegoRol/Mago.java
package Ejercicio3_JuegoRol;

public class Mago extends Personaje {
    private final int poderAtaque = 8;
    private int mana = 50;

    public Mago(String nombre, int puntosVida) {
        super(nombre, puntosVida);
    }

    @Override
    public int atacar() {
        System.out.println(nombre + " lanza un hechizo!");
        return poderAtaque;
    }
    @Override
    public void usarHechizo(Hechizo hechizo) {
        if (mana >= hechizo.getCostoMana()) {
            System.out.println(nombre + " usa " + hechizo.getNombre() + "!");
            System.out.println("Causa " + hechizo.getPoder() + " de daño!");
            mana -= hechizo.getCostoMana(); // Restar el costo de mana
        } else {
            System.out.println(nombre + " no tiene suficiente mana para usar " + hechizo.getNombre() + ".");
        }
    }
}

File: Ejercicios_POO3.1_Kotlin/src/Ejercicio3_JuegoRol/Mago.kt
package Ejercicio3_JuegoRol

class Mago(nombre: String, puntosVida: Int) : Personaje(nombre, puntosVida) {
    private val poderAtaque = 8
    private var mana = 50

    override fun atacar(): Int {
        println("$nombre lanza un hechizo!")
        return poderAtaque
    }
    override fun usarHechizo(hechizo: Hechizo) {
        if (mana >= hechizo.costoMana) {
            println("$nombre usa ${hechizo.nombre}!")
            println("Causa ${hechizo.poder} de daño!")
            mana -= hechizo.costoMana // Restar el costo de mana
        } else {
            println("$nombre no tiene suficiente mana para usar ${hechizo.nombre}.")
        }
    }
}